Sources/NIOCore/BSDSocketAPI.swift

```diff
@@ -199,12 +204,14 @@ extension NIOBSDSocket.ProtocolFamily {
     public static let inet6: NIOBSDSocket.ProtocolFamily =
             NIOBSDSocket.ProtocolFamily(rawValue: PF_INET6)
 
+#if !os(WASI)
```

Comment:
My general preference here would be that the APIs here are _present_, they just end up failing when actually used in WASI. That would mean hardcoding the syscall value on WASI, or deliberately using a spoiler value. That note applies to the rest of this file.

Reply:
What would be the benefit of that as compared to compile-time warnings for absent symbols? My worry is that someone ends up adopting these APIs when targeting WASI and then discovering that they fail at the most inconvenient moment.

If you insist on runtime errors, would you prefer a `fatalError` or a certain sentinel value? In the latter case, how do I pick such value or do we just change `NIOBSDSocket.Option.rawValue` to have `CInt?` type instead of `CInt`?

Reply:
One other argument for compile-time checks is consistency: Windows did it that way, so why WASI shouldn't follow that precedent?